shared/eip712/eip712.go

```diff
@@ -0,0 +1,54 @@
+package eip712
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+)
+
+type EIP712Components struct {
+	R [32]byte `json:"r"`
+	S [32]byte `json:"s"`
+	V uint8    `json:"v"`
+}
+
+const EIP712Length = 65
+
+// Pretty print for EIP712Components
+func (e EIP712Components) String() {
```

Comment:
```suggestion
func (e *EIP712Components) Print() {
```
(renames to Print and changes the receiver to a pointer type to match your other receivers, which is moderately important)